lib/sidekiq/worker_stats/stats.rb

```diff
@@ -83,6 +85,29 @@ def memory_measurement
       def current_memory
         `awk '{ print $2 }' /proc/#{@pid}/statm`.strip.to_i * @page_size
       end
+
+      # Remove old samples if number of samples > @config.max_samples
+      def remove_old_samples
+        keys = nil
+        Sidekiq.redis do |redis|
+          keys = redis.hkeys ::Sidekiq::WorkerStats::REDIS_HASH
+        end
+
+        return if keys.length <= @config.max_samples
```

Comment:
Yeah and could even do it before redis.hkeys and since it only takes O(1) instead of O(N) it could cut before this, will do it on friday along with fixing other issues since it's not crucial.